src/map/achievement.cpp

```diff
@@ -691,54 +691,41 @@ int achievement_check_progress(struct map_session_data *sd, int achievement_id,
  * Calculate a player's achievement level
  * @param sd: Player to check achievement level
  * @param flag: If the call should attempt to give the AG_GOAL_ACHIEVE achievement
- * @return Player's achievement level or 0 on failure
+ * @return Rollover and TNL EXP or 0 on failure
  */
 int *achievement_level(struct map_session_data *sd, bool flag)
 {
-	static int info[2];
-	int i, old_level;
-	const int score_table[MAX_ACHIEVEMENT_RANK] = { 18, 31, 49, 73, 135, 104, 140, 178, 900, 1000, 1100, 1200, 1300, 1400, 1500, 1600, 1700, 1800, 1900, 2000 }; //! TODO: Figure out the EXP required to level up from 8-20
-
 	nullpo_retr(0, sd);
 
+	const int score_table[MAX_ACHIEVEMENT_RANK] = { 18, 31, 49, 73, 135, 104, 140, 178, 900, 1000, 1100, 1200, 1300, 1400, 1500, 1600, 1700, 1800, 1900, 2000 }; //! TODO: Figure out the EXP required to level up from 8-20
+	int info[2], old_level = sd->achievement_data.level, temp_score, temp_level = 0;
```

Comment:
Experiencing a warning..

` warning: address of local variable ‘info’ returned [-Wreturn-local-addr]
  int info[2], old_level = sd->achievement_data.level, temp_score, temp_level = 0;
`